arangod/MMFiles/MMFilesHashIndex.cpp

```diff
@@ -633,7 +633,7 @@ Result MMFilesHashIndex::insertUnique(transaction::Methods* trx,
       }
 
       // Already indexed elements will be removed by the rollback
-      return error;
+      return std::move(error);
```

Comment:
pessimization? disables RVO?

Reply:
RVO is not guarateed here.
clang-8 also complains here.